src/SheetsRegistry.js

```diff
@@ -15,7 +15,22 @@ export default class SheetsRegistry {
    * @api public
    */
   add(sheet) {
-    this.registry.push(sheet)
+    const {registry} = this
+
+    if (!registry.length || typeof sheet.options.index !== 'number') {
+      registry.push(sheet)
+      return
+    }
+
+    for (let i = 0; i < registry.length; i++) {
+      const {options} = registry[i]
+      if (options.index > sheet.options.index) {
+        registry.splice(i, 0, sheet)
+        return
+      }
+    }
+
+    registry.push(sheet)
```

Comment:
@kof the `for` loop will leave it hanging if it belongs at the end.

 We could add an extra conditional if we're on the last item but it hasn't been inserted (so just after line 30). Or instead check the last sheet in the registry beforehand to avoid the iteration all together.

Another option is to use a binary search to find the insertion index, but I don't think there would be any point.